File: backend/ProgrammingInternshipPlatform/ProgrammingInternshipPlatform.Api/InternshipHub/Contracts/Responses/Member.cs
using ProgrammingInternshipPlatform.Application.Abstractions.GraphApi.Responses;

namespace ProgrammingInternshipPlatform.Api.InternshipHub.Contracts.Responses;

public class Member
{
    public Guid MemberId { get; private set; }
    public string DisplayName { get; private set; } = null!;
    public string GivenName { get; private set; } = null!;
    public string Surname { get; private set; } = null!;
    public string Initials { get; private set; } = null!;
    public string JobTitle { get; private set; } = null!;
    public string Email { get; private set; } = null!;
    public string ShortenedEmail { get; private set; } = null!;
    public string? PictureUrl { get; private set; }
    public List<string>? Roles { get; private set; }

    public static Member MapFrom(Account account)
    {
        return new Member
        {
            MemberId = account.Id.Value,
            DisplayName = account.DisplayName,
            GivenName = account.GivenName,
            Surname = account.Surname,
            Initials = account.Initials,
            JobTitle = account.JobTitle,
            Email = account.Email,
            ShortenedEmail = ShortenEmail(account.Email)
        };
    }

    private static  string ShortenEmail(string originalEmail)
    {
        var atIndex = originalEmail.IndexOf("@", StringComparison.Ordinal);
        var emailPrefix = originalEmail.Substring(0, atIndex + 1);
        var newSuffix = "shine.com";
        var shortenedEmail = $"{emailPrefix}{newSuffix}";
        return shortenedEmail;
    }
}

File: shine/src/app/internship-hub/models/member.ts
import { Account, AccountWithPicture } from "src/app/shared/models/account";

export interface MemberServerResponse extends AccountWithPicture {
}

export class Member {
    public coordinatorId: string;
    public displayName: string;
    public givenName: string;
    public surname: string;
    public initials: string;
    public jobTitle: string;
    public email: string;
    public shortenedEmail: string;
    public pictureUrl?: string;
    public role: string;

    private constructor(response: MemberServerResponse) {
        this.coordinatorId = response.coordinatorId;
        this.displayName = response.displayName;
        this.givenName = response.givenName;
        this.surname = response.surname;
        this.initials = response.initials;
        this.jobTitle = response.jobTitle;
        this.email = response.email;
        this.shortenedEmail = response.shortenedEmail;
        this.pictureUrl = response.pictureUrl;
        this.role = response.role;
    }

    public static createFromResponse = (response: MemberServerResponse): Member => 
        new Member(response);
}